### ISSUE TITLE
Partial Update v2

### ISSUE THREAD

**Issue by feeleen** [linq2db/linq2db#2185]
related to #1044 

Maybe linq2db "Update" method could have additional parameter with list of Properties (non-pk) to update?
If I use my own change tracking code to monitor properties that changed - then I want to pass these props to Update method. It's unnessesary to update every property in this case

**Comment by sdanyliv** [comment]
I have written several examples how to do that via reflection.
I can search in issues or in my snippets if you are interested in that smples.

**Comment by feeleen** [comment]
@sdanyliv yes i'm interested in that example, could you post it here please

**Comment by sdanyliv** [comment]
I have found this sample, I think it is a good start:
https://github.com/linq2db/linq2db/issues/1172#issuecomment-394308169

**Comment by sdanyliv** [comment]
It is more about strings, but you can use PropertyInfo with minimal changes.
```cs
var entityDescriptor = MappingSchema.GetEntityDescriptor(type);
```
This descriptor contains all needed information for creating update.

**Comment by feeleen** [comment]
thanks for example, but it seems too complicated for such a simple task.. Also we have to deal with new objects with property values, which is not good.. 

why just not to add extra parameter to update/createquery methods? I don't think this would go against any of linq2db concepts..

for example we add parameter **updatedFields** and have something like this in CreateQuery:

`
var fields = sqlTable.Fields.Values.Where(f => f.IsUpdatable && !f.ColumnDescriptor.ShouldSkip(obj, descriptor, SkipModification.Update) && (updatedFields.Count == 0 || updatedFields.Contains(f.Name))).Except(keys).ToList();


`

**Comment by sdanyliv** [comment]
It is debatable.

Then someone will ask for updating `ModfiiedOn` by current timestamp, another additional api modification.
Then it will be questions: should we use PropertyNames, PropertyInfos, ColumnNames? Or maybe better to expose callback (I like this idea)?

In your case, better to create your extensions, based on my sample. Better because it will work now ;) linq2db gives extensibility in such case.

If you can not solve it yourself, I can help in that.

**Comment by sdanyliv** [comment]
You can create PR with tests. I think it should be overload with callback:
```cs
		public static int Update<T>(
			this IDataContext dataContext,
			T obj,
			Func<object, ColumnDescriptor, bool> columnFilter,
			string? tableName    = null,
			string? databaseName = null,
			string? schemaName   = null,
			string? serverName   = null)
		{
			//
		}


```

**Comment by feeleen** [comment]
@sdanyliv created PR with your callback fn

**Comment by MaceWindu** [comment]
Reopen. PR is not ready and was merged too early #2203

**Comment by MaceWindu** [comment]
merged
